### PR TITLE
Feat/hearder and footer

### PR DESCRIPTION
**Pull request by NataRep**
#### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Test Case
- [ ] Documentation update
- [ ] Other

#### 💡 Background and solution

image

#### 🎯 Related issues

[_Links to related issues SCRUM-17_
](https://angularteam.atlassian.net/browse/SCRUM-17)

### CODE REVIEW

File: src/app/core/store/user/user.actions.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { createAction, props } from '@ngrx/store';
+
+export const signIn = createAction(
+  '[User] User signIn',
+  props<{ role: 'guest' | 'user' }>()
```
Comment (Arthurchkeee) [inline src/app/core/store/user/user.actions.ts:5]: We have 'manager' role too

Comment (Nuttik) [reply src/app/core/store/user/user.actions.ts:5]: Yes, there is a mistake, Admin is missing. 
I took this part of the code from Sasha's PR, I will replace it before the merge.

File: src/app/components/header/menu/menu.component.html [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      Sign Up
+    </button>
+    <button mat-flat-button [routerLink]="this.navLinks.guest.signin">
+      Sign In
+    </button>
+  </ng-container>
+  <ng-container *ngIf="role === 'user'">
+    <button mat-flat-button (click)="onLogout()">Logout</button>
+  </ng-container>
+  <ng-container *ngIf="role === 'admin'">
```
Comment (Arthurchkeee) [inline src/app/components/header/menu/menu.component.html:34]: We don't have 'admin' role, in our app it's 'manager'
image

Comment (Nuttik) [reply src/app/components/header/menu/menu.component.html:34]: Okay, I'll correct the role name everywhere to the current one before the merge.

File: src/app/core/guards/admin.guard.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import { selectCurrentAccess } from '../store/user/user.selectors';
+
+export const adminGuard: CanActivateFn = () => {
+  const store = inject(Store);
+  const router = inject(Router);
+
+  return store.select(selectCurrentAccess).pipe(
+    take(1),
+    map((currentAccess) => {
+      if (currentAccess === 'admin') {
```
Comment (Arthurchkeee) [inline src/app/core/guards/admin.guard.ts:14]: See as above

File: src/app/core/guards/guest.guard.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+export const guestGuard: CanActivateFn = () => {
+  const store = inject(Store);
+  const router = inject(Router);
+
+  return store.select(selectCurrentAccess).pipe(
+    take(1),
+    map((currentAccess) => {
+      if (currentAccess === 'guest') {
+        console.log('гость');
```
Comment (Arthurchkeee) [inline src/app/core/guards/guest.guard.ts:15]: Debugging console.log

**Review by Arthurchkeee — CHANGES_REQUESTED** [review]
Great and adaptivity for phones view! Please, rename 'admin' to 'manager' in marked places

File: src/app/core/guards/admin.guard.ts (hunk 3 above)
Comment (Nuttik) [reply src/app/core/guards/admin.guard.ts]: Done

File: src/app/components/header/menu/menu.component.html (hunk 2 above)
Comment (Nuttik) [reply src/app/components/header/menu/menu.component.html]: Done

File: src/app/components/header/header.component.ts [hunk 5]
```diff
@@ -1,10 +1,33 @@
...
 
 @Component({
   selector: 'app-header',
   templateUrl: './header.component.html',
-  styleUrl: './header.component.scss'
+  styleUrl: './header.component.scss',
 })
-export class HeaderComponent {
+export class HeaderComponent implements OnInit {
+  role$!: Observable<Access>;
```
Comment (Aleksa013) [inline src/app/components/header/header.component.ts:15]: I think here need to add access modifiers

File: src/app/components/header/header.component.ts [hunk 6]
```diff
@@ -1,10 +1,33 @@
...
 
+  isMenuOpen: boolean = false;
+
+  // eslint-disable-next-line @ngrx/no-typed-global-store
```
Comment (Aleksa013) [inline src/app/components/header/header.component.ts:19]: may be fix settings?I heard that committing a line is not the best practice

File: src/app/components/header/menu/menu.component.html (hunk 2 above)
Comment (Aleksa013) [reply src/app/components/header/menu/menu.component.html]: Also had this error)))

File: src/app/core/store/user/user.actions.ts (hunk 1 above)
Comment (Aleksa013) [reply src/app/core/store/user/user.actions.ts:5]: I alredy replace this on guest| user| manager

**Review by Aleksa013 — COMMENTED** [review]
Great job!
Please, add access modifiers(public-private-protected)

File: src/app/components/header/header.component.ts (hunk 6 above)
Comment (mogitrash) [reply src/app/components/header/header.component.ts:19]: 👍

File: src/app/components/header/header.component.ts (hunk 5 above)
Comment (mogitrash) [reply src/app/components/header/header.component.ts:15]: yeah, i think it's will be better solution

File: src/app/components/header/menu/menu.component.ts [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+  @Input() onLogout!: () => void;
+
+  @Input() isMobile!: boolean;
+
+  navLinks = {
+    guest: { home: '/', signup: '/signup', signin: '/signin' },
+    user: { profile: '/profile', orders: '/my-orders' },
+    admin: { profile: '/profile', orders: '/orders', panel: '/admin' },
+  };
```
Comment (mogitrash) [inline src/app/components/header/menu/menu.component.ts:19]: good solution 💯

**Review by mogitrash — CHANGES_REQUESTED** [review]
Good job and perfect visual! Have some suggestions)

**Review by Arthurchkeee — APPROVED** [review]

File: src/app/components/header/header.component.ts (hunk 5 above)
Comment (Nuttik) [reply src/app/components/header/header.component.ts]: Done

**Review by Arthurchkeee — APPROVED** [review]

File: src/app/components/header/header.component.ts (hunk 6 above)
Comment (Nuttik) [reply src/app/components/header/header.component.ts:19]: Done

File: src/app/core/store/user/user.actions.ts (hunk 1 above)
Comment (Nuttik) [reply src/app/core/store/user/user.actions.ts]: Done

File: src/app/components/header/menu/menu.component.ts (hunk 7 above)
Comment (Nuttik) [reply src/app/components/header/menu/menu.component.ts]: Thank you!

**Review by mogitrash — APPROVED** [review]
🥇

**Review by Aleksa013 — APPROVED** [review]

**Pull request merged**